google/stackdriver_monitoring/spectator_client.py

```diff
@@ -151,12 +160,67 @@ def __init__(self, options):
     self.__host = options['host']
     self.__prototype = None
     self.__default_scan_params = {'tagNameRegex': '.+'}
+    self.__previous_scan_lock = threading.Lock()
+    self.__previous_scan = {} if options.get('log_metric_diff') else None
 
     if options['prototype_path']:
       # pylint: disable=invalid-name
       with open(options['prototype_path']) as fd:
         self.__prototype = json.JSONDecoder().decode(fd.read())
 
+  def __log_scan_diff(self, host, port, metrics):
+    """Diff this scan with the previous one for debugging purposes."""
+    if self.__previous_scan is None:
+      return
+
+    key = '{0}:{1}'.format(host, port)
+    with self.__previous_scan_lock:
+      previous_metrics = self.__previous_scan.get(key, {})
+      self.__previous_scan[key] = copy.deepcopy(metrics)
+    if not previous_metrics:
+      return
+
+    previous_keys = set(previous_metrics.keys())
+    keys = set(metrics.keys())
+    new_keys = keys.difference(previous_keys)
+    same_keys = keys.intersection(previous_keys)
+    lost_keys = previous_keys.difference(keys)
+    lines = []
+
+    if lost_keys:
+      lines.append('Lost metrics for:\n  - {0}\n'
```

Comment:
What does 'Lost metrics' mean? It makes sense where the `lost_keys` come from, but the terminology seems weird.